### ISSUE TITLE
can we add an effect on hover?

### ISSUE THREAD

**Issue by voidale** [fians/Waves#105]
I think it will be really nice because I have some instant buttons like hrefs and effect on them is practically useless because it's redirect instantly...

**Comment by fians** [comment]
I have no plan to do that. But I will work on it if I have enough time.

**Comment by hoxxep** [comment]
@voidale Is pull request #110 sort of what you meant? So when you drag the mouse over an element it has a rippling/skimming effect like dragging your finger on the surface of water?

If you haven't cloned/forked the repo, I'll comment here if it gets included in the v0.7.0 [dev] build so you can test it.

**Comment by fians** [comment]
Sorry, the drag effect is too complicated. It should be like click/touch event, the difference it will use `mouseenter` and `mouseleave`, so when mouse enter the element, ripple is created (just once), and it will disappear as soon as mouse leave the element.

**Comment by hoxxep** [comment]
So we should add another event alongside dragging, but works on `mouseenter`? Would it make the ripple start from the centre, or from the mouse position on entering the element?

**Comment by hoxxep** [comment]
Now possible to set up hover effects by using `Waves.ripple(elements, options)` with `mouseenter` and `mouseleave` Event Listeners. For example:

```javascript
$('#id').mouseenter(function() {
    Waves.ripple('#id', {wait: null});
}).mouseleave(function() {
    Waves.calm('#id');
});
```

We could make `elements` (from `Waves.ripple(elements)`) also accept raw DOM elements, so that the following is possible:

```javascript
$('#id').mouseenter(function() {
    Waves.ripple(this, {wait: null});
}).mouseleave(function() {
    Waves.calm(this);
});
```

**Comment by voidale** [comment]
Isn't it possible to do with css (hover) only? @hoxxep  it's great if it works I will test later.

**Comment by hoxxep** [comment]
Do you mean using the ripple effect on hover, or just a minor CSS colour change/effect on hover like so?

```css
.waves-button:hover {
    background: blue;
}
```

**Comment by voidale** [comment]
@hoxxep I was wondering, If we really have to use jQuery for the ripple effect. Can't be done using css hover? i guess not. Doesn't matter it's great thanks for the new option! I think it will be useful for many users. Also saw your

**Comment by hoxxep** [comment]
It sadly can't be done through CSS since the ripple must be added via JS, sorry! But thanks for sparking the idea for giving users JS control of the ripples :smile:

It'll be released with v0.7.0, which will hopefully ship within the next few weeks.
